python/cudf/cudf/tests/test_joining.py

```diff
@@ -735,3 +735,56 @@ def test_join_empty_table_dtype():
     pd_merge = left.merge(right, how="left", left_on=["a"], right_on=["b"])
     gd_merge = gleft.merge(gright, how="left", left_on=["a"], right_on=["b"])
     assert_eq(pd_merge["a"].dtype, gd_merge["a"].dtype)
+
+
+@pytest.mark.parametrize("how", ["outer", "inner", "left", "right"])
+@pytest.mark.parametrize("cat", [True, False])
+def test_join_multi(how, cat):
+
+    size = 10
```

Comment:
I'm uncomfortable with this small of a size, as we're using it to produce values for exact-match, and it has a pretty good likelihood to not produce a given value. Actually, I'd prefer at least part of the test data to be declared explicitly for the same reason. If I'm not mistaken, the odds of `np.arange(size)` producing overlapping results for both cudf and pandas dataframes is extremely low, even for high-ish values of `size`. tldr; I'd like to see more-deterministic tests.

Reply:
> ...the odds of np.arange(size) producing overlapping results for both cudf and pandas dataframes is extremely low...

I agree that more-deterministic tests would be ideal.  Hoever, I'm not really sure what you mean by this.  either way - I'll try to make improvements here.

Reply:
I made some erroneous comments. ^^^

`np.arange(size)` doesn't produce random results. My bad.

`np.random.randint(0, 2, size=size)` will likely produce at least one of each value in the range, but it's possible it won't. I'd like to see explicit test data (not random) that will test overlap + lack of overlap on both the left and right sides, and then maybe some random data for good measure.

I see us testing random values against random values in lots of places, and while that's great for testing arithmetic, it's not deterministic when testing discreet value comparisons.